sys/include/ztimer/xtimer_compat.h

```diff
@@ -71,7 +71,32 @@ static inline uint64_t xtimer_now_usec64(void)
 static inline void xtimer_sleep(uint32_t seconds)
 {
     /* TODO: use ZTIMER_SEC */
-    ztimer_sleep(ZTIMER_USEC, seconds * 1000000LU);
+    if (IS_ACTIVE(MODULE_ZTIMER_MSEC)) {
+        while (seconds > (UINT32_MAX / 1000LU)) {
+            ztimer_sleep(ZTIMER_MSEC, UINT32_MAX);
+            seconds -= UINT32_MAX / 1000LU;
+        }
+        ztimer_sleep(ZTIMER_MSEC, seconds * 1000LU);
+    } else {
+        while (seconds > (UINT32_MAX / 1000000LU)) {
+            ztimer_sleep(ZTIMER_USEC, UINT32_MAX);
+            seconds -= UINT32_MAX / 1000000LU;
+        }
+        ztimer_sleep(ZTIMER_USEC, seconds * 1000000LU);
+    }
+}
+
+static inline void xtimer_msleep(uint32_t milliseconds)
+{
+    if (IS_ACTIVE(MODULE_ZTIMER_MSEC)) {
+        ztimer_sleep(ZTIMER_MSEC, milliseconds);
+    } else {
+        while (milliseconds > (UINT32_MAX / 1000LU)) {
+            ztimer_sleep(ZTIMER_USEC, UINT32_MAX);
+            seconds -= UINT32_MAX / 1000LU;
+        }
+        ztimer_sleep(ZTIMER_USEC, milliseconds * 1000LU);
+    }
```

Comment:
This would IMO now justify an `ztimer_msleep()` as helper - for when the user really doesn't care on which hardware the timer is set. But that should be another discussion in another PR.

Reply:
why not just depend on ztimer msec?

Reply:
the helper would do the intermediate wakeups N times for N ztimer_msleep instances. `ztimer_sleep(ZTIMER_MSEC, ...` would mitigate that.

Reply:
Hmm, is there a use case for using `ztimer` without `ztimer_msec`?

Reply:
well, if nothing uses it, no need to include it. but that's not the point here, there's no case (or there should not be one) where ztime_msec is not available.